doc/dev/custom_devices.rst

```diff
@@ -17,91 +17,116 @@ Additionally, all measurements will always return ``true``.
-                                         const std::vector<QubitIdType> &, bool)
+            void MatrixOperation(const std::vector<std::complex<double>> &,
+                                        const std::vector<QubitIdType> &, bool) override
             {
             }
 
-            virtual auto Expval(ObsIdType) -> double { return 0.0; }
-            virtual auto Var(ObsIdType) -> double { return 0.0; }
-            virtual void State(DataView<std::complex<double>, 1> &) {}
-            virtual void Probs(DataView<double, 1> &) {}
-            virtual void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) {}
-            virtual void Sample(DataView<double, 2> &, size_t) {}
-            virtual void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) {}
-            virtual void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) {}
-
-            virtual void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
-                                       const std::vector<QubitIdType> &, size_t)
+            auto Expval(ObsIdType) -> double override { return 0.0; }
+            auto Var(ObsIdType) -> double override { return 0.0; }
+            void State(DataView<std::complex<double>, 1> &) override {}
+            void Probs(DataView<double, 1> &) override {}
+            void PartialProbs(DataView<double, 1> &, const std::vector<QubitIdType> &) override {}
+            void Sample(DataView<double, 2> &, size_t) override {}
+            void PartialSample(DataView<double, 2> &, const std::vector<QubitIdType> &, size_t) override {}
+            void Counts(DataView<double, 1> &, DataView<int64_t, 1> &, size_t) override {}
+
+            void PartialCounts(DataView<double, 1> &, DataView<int64_t, 1> &,
+                                    const std::vector<QubitIdType> &, size_t) override
             {
             }
 
-            virtual void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) {}
+            void Gradient(std::vector<DataView<double, 1>> &, const std::vector<size_t> &) override {}
         };
 
-In addition to implementing the ``QuantumDevice`` class, one must implement the following method:
+In addition to implementing the ``QuantumDevice`` class, one must implement the following method
+with the name ``CustomDevice`` + ``Factory``:
 
 .. code-block:: c++
 
     extern "C" Catalyst::Runtime::QuantumDevice*
-    getCustomDevice() { return new CustomDevice(); }
+    CustomDeviceFactory(bool status, const std::string &kwargs)
+    {
+        return new CustomDevice(status, kwargs); }
+
+The use of the term ``Factory`` in ``CustomDeviceFactory`` is a runtime convention that combines the name
+of the device class, in this case, ``CustomDevice``, with the suffix ``Factory``. This signifies that the
+primary purpose of the method is to act as a factory for creating instances of the ``CustomDevice`` class.
```

Comment:
```suggestion
The entry point function acts as a factory method for the device class. Note that a plugin library
may also provide several factory methods in case it packages multiple devices into the same
library. However, it is important that the device identifier be unique, as best as possible, to
avoid clashes with other plugins. 
```

Reply:
Updated with some changes [e425745](https://github.com/PennyLaneAI/catalyst/pull/343/commits/e42574552ea513430058dec8fee1bb6392071720)